node_test.go

```diff
@@ -57,149 +58,174 @@ func (s *DecisionSuite) Test(c *C) {
 			`device.age_group != "13-17"`,
 			`geo_code matches '^(.*,)?(US)$'`,
 		},
+		"object07": []string{
+			`geo_code matches '^(.*,)?(US)$'`,
+			`platform = 'iOS'`,
+			`device.gender != "male"`,
+		},
 	}
 
-	tree, err := NewTree(objects)
+	tree, err := NewTree(s.Objects)
 	if err != nil {
 		c.Fatal(err)
 	}
 
-	//fmt.Println(tree)
-
 	f, err := os.Create("decision.dot")
 	if err != nil {
 		c.Fatal(err)
 	}
 	defer f.Close()
 	tree.Graph(f)
 
+	s.Tree = tree
+}
+
+func (s *DecisionSuite) Test_Context1(c *C) {
 	context := TestContext{
 		"geo_code":         "US",
 		"platform":         "iOS",
 		"device.age_group": "60",
 	}
 
-	logger := &TestLogger{
-		make([]string, 0, 10),
+	s.testEvaluate(c, context, []string{
+		"object01",
+		"object03",
+	})
+}
+
+func (s *DecisionSuite) Test_Context2(c *C) {
+	context := TestContext{
+		"geo_code": "US",
 	}
+	s.testEvaluate(c, context, []string{
+		"object01",
+	})
+}
 
-	log, err := tree.Evaluate(context, logger)
-	if err != nil {
-		c.Fatal(err)
+func (s *DecisionSuite) Test_Context3(c *C) {
+	context := TestContext{
+		"geo_code": "US",
+		"platform": "iOS",
 	}
+	s.testEvaluate(c, context, []string{
+		"object01",
+		"object03",
+	})
+}
 
-	c.Assert(logger.Trace, HasLen, 4)
+func (s *DecisionSuite) Test_Context4(c *C) {
+	context := TestContext{
+		"geo_code":      "US",
+		"platform":      "iOS",
+		"device.gender": "female",
+	}
+	s.testEvaluate(c, context, []string{
+		"object01",
+		"object03",
+		"object07",
+	})
+}
 
-	c.Log(log)
+func (s *DecisionSuite) Test_Context5(c *C) {
+	context := TestContext{
+		"geo_code":         "US",
+		"platform":         "iOS",
+		"device.age_group": "60",
+	}
+	s.testEvaluate(c, context, []string{
```

Comment:
I think that since the next logical step asks if ```device.gender != 'male'```` we should also be returning object07 since it seems if gender is not specified, the request could be treated as having any gender